envi/archs/msp430/__init__.py

```diff
@@ -1,52 +1,50 @@
 """
-The initial msp430 module.
+msp430 module
 """
+
 ############
 # Author: Don C. Weber
 # Started: 05/23/2009
 # 
 
-# TODO:
-#
-
 import envi
 
-import disasm
+from envi.archs.msp430.regs import *
+from envi.archs.msp430.disasm import *
+from envi.archs.msp430.const import *
 
 class Msp430Module(envi.ArchitectureModule):
 
     def __init__(self):
         envi.ArchitectureModule.__init__(self, "msp430", maxinst=4)
-        self._arch_dis = disasm.Msp430Disasm()
+        self._arch_dis = Msp430Disasm()
 
     def archGetRegCtx(self):
-        return regs.Msp430RegisterContext()
-
-    def archGetBreakInstr(self):
-        return '\xcc'
+        return Msp430RegisterContext()
 
     def archGetNopInstr(self):
-        return '\x90'
+        return '\x03\x43' # NOP is emulated with: MOV #0, R3
 
     def archGetRegisterGroups(self):
         groups = envi.ArchitectureModule.archGetRegisterGroups(self)
-
-        groups.append(regs.GeneralRegGroup)
+        general= ('general', registers, )
+        groups.append(general)
         return groups
 
     def getPointerSize(self):
-        return 4
+        return WORD
```

Comment:
i think getPointerSize wants to be a number, period.  WORD is yet another indirection and it doesn't really gain us anything.